source/blog/2017-05-17-some-lessons-an-it-department-can-learn-from-openstack.html.md

```diff
@@ -0,0 +1,74 @@
+---
+title: Some lessons an IT department can learn from OpenStack
+author: jpena
+tags: 
+date: 2017-05-17 13:39:19 CEST
+---
+
+I have spent a lot of my professional carrer working as an IT Consultant/Architect. In those positions, you talk to many customers with different backgrounds, and see companies that run their IT in many different ways. Back in 2014, I joined the OpenStack Engineering team at Red Hat, and started being involved with the OpenStack community. And guess what, I found yet another way of managing IT.
+
+These last 3 years have taught me a lot about how to efficiently run an IT infrastructure at scale, and what's better, they proved that many of the concepts I had been previously preaching to customers (_automate, automate, automate!_) are not only viable, but also required to handle ever-growing requirements with a limited team and budget.
+
+So, would you like to know what I have learnt so far in this 3-year journey?
+
+## Processes
+
+The OpenStack community relies on several processes to develop a _cloud operating system_. Most of these processes have evolved over time, and together they allow a very large contributor base to collaborate effectively. Also, we need to manage a complex infrastructure to support this our processes.
+
+- **Infrastructure as code**: there are several important servers in the OpenStack infrastructure, providing service to thousands of users every day: the Git repositories, the Gerrit code review infrastructure, the CI bits, etc. The deployment and configuration of all those pieces is automated, as you would expect, and the Puppet modules and Ansible playbooks used to do so are available at their [Git repository](https://github.com/openstack-infra/system-config). There can be no snowflakes, no "this server requires a very specific configuration, so I have to log on and do it manually" excuses. If it cannot be automated, it is not efficient enough. Also, storing our infrastructure definitions as code allows us to take changes through peer-review and CI before applying in production. More about that later.
+
+- **Development practices**: each OpenStack project follows the same structure:
+
+  * There is a Project Team Leader (PTL), elected from the project contributors every six months. A PTL acts as a project coordinator, rather than a manager in the traditional sense, and is usually expected to rotate every few cycles.
+  * There are several core reviewers, people with enough knowledge on the project to judge if a change is correct or not.
+  * And then we have multiple project contributors, who can create patches and peer-review other people's patches.
+
+  Whenever a patch is created, it is sent to review using a code review system, and then:
+
+  * It is checked by multiple CI jobs, that ensure the patch is not breaking any existing functionality.
+  * It is reviewed by other contributors.
```

Comment:
I'd reinforce the importance of peer review process by explaining its advantages, as enforcing people justify and document changes, enforce some policies, it's very good for newcomers, etc... just an idea, you could create subsections for peer-review and gating CI with a bit longer explanation.